packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -119,7 +119,35 @@ const DEFAULT_BASE_OPTIONS = Object.freeze({
 });
 
 /**
- * Encapsulates logic for the service provider for the mongosh CLI.
+ * Connect a MongoClient. If AutoEncryption is requested, first connect without the encryption options and verify that
+ * the connection is to an enterprise cluster. If not, then error, otherwise close the connection and reconnect with the
+ * options the user initially specified. Provide the client class as an additional argument in order to test.
+ * @param uri {String}
+ * @param clientOptions {MongoClientOptions}
+ * @param mClient {MongoClient}
+ */
+export async function connectMongoClient(uri: string, clientOptions: MongoClientOptions, mClient = MongoClient): Promise<MongoClient | void> {
```

Comment:
When would this not return a `Promise<MongoClient>`?

Reply:
Yes, it would! Good catch - I previously had it looking like
```
if (x) {
    return y;
} 
throw z
```
which I just learned you can't have the function signature return just the type of `y`. Then I refactored to avoid that issue, but then forgot to update the signature  🤦